fgpyo/util/metric.py

```diff
@@ -148,18 +148,25 @@ def _parsers(cls) -> Dict[type, Callable[[str], Any]]:
         return {}
 
     @classmethod
-    def read(cls, path: Path) -> Iterator[Any]:
+    def read(cls, path: Path, skip_extra: bool = True) -> Iterator[Any]:
         """Reads in zero or more metrics from the given path.
 
         The metric file must contain a matching header.
 
         Args:
             path: the path to the metrics file.
+            skip_extra: True to ignore any extra columns, False to raise an exception.
         """
         parsers = cls._parsers()
         with path.open("r") as reader:
             header: List[str] = reader.readline().rstrip("\r\n").split("\t")
-            assert header == cls.header(), "Header did not match"
+            cls_header = cls.header()
+            # check the header
+            for field in cls_header:
+                assert field in header, f"Missing field '{field}' in file: {path}"
```

Comment:
I would _love_ it if this check was also allowed (optionally?) to skip over situations where a) the class has a field, b) it's not in the file, c) the field is `Optional[_]` in the class.  I.e. would be nice to be able to evolve classes adding optional fields, and be able to parse old files.

Reply:
I love that idea

Reply:
I did this!